File: android/app/src/main/java/net/quakemesh/android/transport/Transport.kt
// Copyright (c) 2026 John Dovey <dovey.john@gmail.com>
//
// Changelog:
//   0.0.5 - Phase 4: transport interface plugged in by Android shims.

package net.quakemesh.android.transport

interface Transport {
    val name: String
    fun start()
    fun stop()
    fun send(peerHex: String, frame: ByteArray)
}

File: core/transport/transport.go
// Copyright (c) 2026 John Dovey <dovey.john@gmail.com>
//
// Changelog:
//   0.0.1 - Initial scaffold.

// Package transport defines the abstract link interface mesh-core depends
// on. Platform-specific implementations (BLE, Wi-Fi Direct, Wi-Fi Aware,
// LAN, internet fallback) are plugged in by the host binary or the Android
// Kotlin layer; mesh-core itself never references a concrete transport.
package transport

// PeerID identifies the far end of a link, scoped to a single Transport.
type PeerID []byte

// Transport is the interface every link implementation must satisfy.
type Transport interface {
	Send(peer PeerID, frame []byte) error
	OnReceive(handler func(peer PeerID, frame []byte))
	OnPeerUp(handler func(peer PeerID))
	OnPeerDown(handler func(peer PeerID))
}
